src/dev/metadata.rs

```diff
@@ -1,7 +1,38 @@
 //! Metadata about circuits.
 
+use crate::plonk::{self, Any};
 use std::fmt;
 
+/// Metadata about a column within a circuit.
+#[derive(Debug, PartialEq)]
+pub struct Column {
+    /// The index of the column.
+    index: usize,
+    /// The type of the column.
+    column_type: Any,
+}
+
+impl fmt::Display for Column {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Column {} ('{:?}')", self.index, self.column_type)
+    }
+}
+
+impl From<(usize, Any)> for Column {
+    fn from((index, column_type): (usize, Any)) -> Self {
```

Comment:
Column type has higher precedence than column index:
```suggestion
impl From<(Any, usize)> for Column {
    fn from((column_type, index): (Any, usize)) -> Self {
```